packages/@react-aria/selection/src/useSelectableItem.ts

```diff
@@ -140,10 +140,19 @@ export function useSelectableItem(options: SelectableItemOptions): SelectableIte
     };
   }
 
-  let modality = useRef(null);
-  let hasPrimaryAction = onAction && manager.selectionMode === 'none';
+
+  // With checkbox selection, onAction (i.e. navigation) becomes primary, and occurs on a single click of the row.
+  // Clicking the checkbox enters selection mode, after which clicking anywhere on any row toggles selection for that row.
```

Comment:
This might be because I'm so used to the old behavior, but this feels pretty unintuitive tbh. Not sure an end user would be able to understand the selection mode switching that happens on checkbox press especially. It feels like onAction should remain as the secondary action with checkbox selection but I understand that having selection remain as the primary action would result in a weird experience (how would you trigger onAction on this case? Double clicking would toggle selection which is definitely weird).

Is there an example design/product that follows this interaction pattern I could look at? I guess Outlook behaves similarly to this? It almost feels like highlight selection should be the go to mode and checkbox selection should strictly disallow onAction...

Reply:
Honestly maybe double clicking to trigger onAction when in checkbox selection wouldn't be that bad. At least the user experience between checkbox selection and highlight selection would be the same